### PR TITLE
HW-655 Adding support for ESP8266 HW-655 Relay board

### PR DESCRIPTION
**Pull request by daniftodi**
Adding support for Esp8266 HW-655 board.

hw-655

### CODE REVIEW

File: code/espurna/config/hardware.h [hunk 1]
```diff
@@ -4059,6 +4059,20 @@
...
+
+    #define MANUFACTURER            "GENERIC"
+    #define DEVICE                  "HW655_RELAY"
+    #define RELAY_PROVIDER          RELAY_PROVIDER_HW655
+    #define SERIAL_BAUDRATE         9600
+
+    // Relays
+    #define DUMMY_RELAY_COUNT             1
+    #define RELAY1_DELAY_ON               0
+    #define RELAY1_DELAY_OFF              0
```
Comment (daniftodi) [inline code/espurna/config/hardware.h]: observed they are defined for other relay board, are they necessary?

File: code/espurna/relay.ino [hunk 2]
```diff
@@ -98,6 +98,16 @@ void _relayProviderStatus(unsigned char id, bool status) {
         rfbStatus(id, status);
     #endif
 
+    #if RELAY_PROVIDER == RELAY_PROVIDER_HW655
+            Serial.write(0xA0);
+            Serial.write(0x01);
+            Serial.write(status);
+            Serial.write(0xA1 + status);
+            Serial.flush();
```
Comment (daniftodi) [inline code/espurna/relay.ino]: A0 A1 00 A1 - close
A0 A1 01 A2 - open

File: code/espurna/system.ino [hunk 3]
```diff
@@ -243,7 +243,7 @@ void _systemSetupSpecificHardware() {
 
     // These devices use the hardware UART
     // to communicate to secondary microcontrollers
-    #if (RF_SUPPORT && !RFB_DIRECT) || (RELAY_PROVIDER == RELAY_PROVIDER_DUAL) || (RELAY_PROVIDER == RELAY_PROVIDER_STM)
+    #if (RF_SUPPORT && !RFB_DIRECT) || (RELAY_PROVIDER == RELAY_PROVIDER_DUAL) || (RELAY_PROVIDER == RELAY_PROVIDER_STM) || (RELAY_PROVIDER == RELAY_PROVIDER_HW655)
```
Comment (daniftodi) [inline code/espurna/system.ino]: we need Serial communication initialized (control of HW-655 relays is done through serial port)

Comment (mcspr) [reply code/espurna/system.ino]: Yes, this should probably be a runtime flag, but we never got around managing debugging like that bc that depends on serial begin too (and reverse-depends on this condition)

File: code/espurna/config/hardware.h (hunk 1 above)
Comment (mcspr) [reply code/espurna/config/hardware.h]: Other one? 0 is the default delay, so this does nothing

File: code/espurna/relay.ino (hunk 2 above)
Comment (mcspr) [reply code/espurna/relay.ino]: STM relay protocol strikes again... slightly different each time

**Comment by daniftodi** [comment]
@xoseperez could you please review this?

**Comment by mcspr** [comment]
if merged, should depend on https://github.com/xoseperez/espurna/issues/713

just to be clear, this is *not* the same as STM relay protocol? i do remember another issue dealing with a similar board, but was resolved as an user error
https://github.com/xoseperez/espurna/issues/1130#issuecomment-482198341

**Comment by daniftodi** [comment]
It works with this STM commands, probably it is another one. not sure.

**Comment by mcspr** [comment]
Can you please try?

Based on the referenced issue, it should be something like this:
```cpp
...
#elif defined(GENERIC_HW_655)

    #define MANUFACTURER            "GENERIC"
    #define DEVICE                  "HW_655"

    #define DUMMY_RELAY_COUNT       1
    #define RELAY_PROVIDER          RELAY_PROVIDER_STM

    #define DEBUG_SERIAL_SUPPORT    0
...
```

**Comment by mcspr** [comment]
>  not sure.

?

**Pull request closed without merging**